webv2/web.go

```diff
@@ -327,6 +340,86 @@ func setSourceDBDetailsForDump(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+//getSourceProfileConfig returns the configured source profile by the user
+func getSourceProfileConfig(w http.ResponseWriter, r *http.Request) {
+	sessionState := session.GetSessionState()
+	sourceProfileConfig := sessionState.SourceProfileConfig
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(sourceProfileConfig)
+}
+
+func setShardsSourceDBDetailsForDataflow(w http.ResponseWriter, r *http.Request) {
+	//Take the received object and store it into session state.
+	sessionState := session.GetSessionState()
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var srcConfig shardedDataflowConfig
+	err = json.Unmarshal(reqBody, &srcConfig)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	sessionState.SourceProfileConfig = srcConfig.MigrationProfile
+	w.WriteHeader(http.StatusOK)
+}
+
+func setShardsSourceDBDetailsForBulk(w http.ResponseWriter, r *http.Request) {
+	sessionState := session.GetSessionState()
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var shardConfigs driverConfigs
+	err = json.Unmarshal(reqBody, &shardConfigs)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	var connDetailsList []profiles.DirectConnectionConfig
+	for i, config := range shardConfigs.DbConfigs {
+		dataSourceName := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", config.User, config.Password, config.Host, config.Port, config.Database)
+		sourceDB, err := sql.Open(config.Driver, dataSourceName)
+		if err != nil {
+			http.Error(w, "Database connection error, check connection properties.", http.StatusInternalServerError)
+			return
+		}
+		// Open doesn't open a connection. Validate database connection.
+		err = sourceDB.Ping()
+		if err != nil {
+			http.Error(w, "Database connection error, check connection properties.", http.StatusInternalServerError)
+			return
+		}
+		sessionState.DbName = config.Database
+		sessionState.SessionFile = ""
+		connDetail := profiles.DirectConnectionConfig{
+			Host:     config.Host,
+			Port:     config.Port,
+			User:     config.User,
+			Password: config.Password,
+			DbName:   config.Database,
+		}
+		connDetailsList = append(connDetailsList, connDetail)
+		//set the first shard as the schema shard when restoring from a session file
+		if shardConfigs.IsRestoredSession == "sessionFile" {
```

Comment:
should use a constant in these checks rather than a string.

Reply:
Done